packages/playwright-core/src/server/dispatchers/dispatcher.ts

```diff
@@ -326,3 +326,16 @@ function formatLogRecording(log: string[]): string {
   const rightLength = headerLength - header.length - leftLength;
   return `\n${'='.repeat(leftLength)}${header}${'='.repeat(rightLength)}\n${log.join('\n')}\n${'='.repeat(headerLength)}`;
 }
+
+function maybeMaskSensitiveData(params: any) {
+  if (!params || !isObject(params))
+    return params;
+  if (!(params as any).secret)
+    return params;
+  const redacted: any = { ...params };
+  for (const name of ['text', 'value']) {
```

Comment:
Let's make this declarative? Like we have `tracing: snapshot: true` in the `protocol.yml`.